src/test/hooks.ts

```diff
@@ -0,0 +1,629 @@
+import * as H from './helper'
+import { expect } from 'chai'
+import { TinyPg } from '../'
+import * as AsyncHooks from 'async_hooks'
+import { TinyHooks } from '../types'
+import * as _ from 'lodash'
+
+// forces PromiseHooks to be enabled.
+AsyncHooks.createHook({ init() {} }).enable()
+
+describe('Hooks', () => {
```

Comment:
some tests around the invalid values to pass as a hook (e.g. `null` and `something other than a function`) would be helpful to ensure that it behaves like a no-op instead of blowing up.